finetune_adapter.py

```diff
@@ -32,6 +32,8 @@
 max_seq_length = 256  # see scripts/prepare_alpaca.py
 warmup_steps = epoch_size * 2 // micro_batch_size // devices  # 2 epochs
 
+data_padded = False
```

Comment:
Thanks for the review @awaelchli. `block_size` is defined as `max_seq_length`. Since the `max_seq_length` is constant, doesn't it make sense to just preprocess the samples so they are all padded to the `max_seq_length` before training rather than padding them to `max_seq_length` each time we gather & prepare a batch?

Reply:
This closely follows the nano-gpt style of data loading. We generally favor simplicity over highly optimized code. I believe the flexibility of being able to choose the max_seq_length and batch size in the script directly is very valuable and needed for users who want to get these scripts running on consumer hardware. We're not seeing any noticable data loading bottlenecks and therefore keep the current simple logic with memory mapping. 

I believe the padding to the constant shape should avoid the re-compilation, is that correct?

Reply:
Yes, padding to a constant shape should avoid the re-compilation. I can revert the changes I made, and instead change the padding to a constant shape if using an xla device,